### PR TITLE
test(parse): pin parse() numeric/bool struct-field defaults + str() formatting

### PR DESCRIPTION
**Pull request by javimosch**
Automated maintenance run by automaintainer.

**Focus:** 

----
WORK ALREADY IN FLIGHT — do not overlap:
These automaintainer pull requests are already open and awaiting review. Do NOT re-implement, refactor, or restructure the files they touch — pick non-overlapping work, and never recreate a change an open PR already makes. If your objective unavoidably overlaps one of these, choose a different, complementary improvement instead.

- PR #456 (am/am-7b5889-ti424g): test(parser): pin parseFuncLit closure-literal parse-error branches
    touches: math_sign_contracts_test.go, parse_funclit_error_test.go
- PR #455 (am/am-7b5889-ti408k): test(builtins): pin cos/tan numeric contracts + even/odd symmetry
    touches: loop_control_flow_test.go, trig_cos_tan_test.go
- PR #454 (am/am-7b5889-ti3y9i): test(builtins): pin trim() whitespace-class + empty/one-sided edges
    touches: main.go, tighten_test.go, trim_edge_test.go
- PR #452 (am/am-7b5889-ti3v45): test(encode): pin brace-counting through strings with braces + escaped quotes
    touches: encode_brace_string_test.go
- PR #450 (am/am-7b5889-ti3t65): [needs work] fix(codegen): parse() must default absent struct string fields to "" not NULL
    touches: CHANGELOG.md, codegen.go, parse_null_string_field_test.go, parse_unset_field_paths_test.go
- PR #446 (am/am-7b5889-ti1vmu): [needs work] feat(examples): pure-MFL decode-step kernels (rmsnorm + stable softmax + argmax)
    touches: decode_step_example_test.go, examples/complex/decode_step.mfl, examples/complex/ffn_activation.mfl, ffn_activation_example_test.go
- PR #443 (am/am-7b5889-ti1pfj): [needs work] docs(examples): Q8_0 quantized GEMV example + pinned-output test
    touches: examples/README.md, examples/complex/quant_matmul.mfl, quant_matmul_example_test.go

**Branch:** `am/am-7b5889-ti43vs`

**Diff:**
```
exp_log_contracts_test.go            | 118 +++++++++++++++++++++++++++++++++++
 parse_numeric_field_defaults_test.go |  44 +++++++++++++
 2 files changed, 162 insertions(+)
```

## Summary by CodeRabbit

* **Tests**
  * Added end-to-end coverage for exponential and logarithmic calculations, including inverse relationships, arithmetic identities, expected values, and monotonicity.
  * Added validation that softmax remains stable when logits are shifted and that probabilities and cross-entropy values are correct.
  * Added coverage for numeric and boolean JSON parsing defaults, value preservation, and numeric string formatting.

### CODE REVIEW

**Comment by javimosch** [comment]
⚠️ **Automaintainer intent check: OFF-OBJECTIVE**

Wait — let me reconsider against the actual objective.

The verification gate passed, but this change may not address the stated objective. Review before merging.

**Comment by javimosch** [comment]
**Automaintainer Review Verdict:** APPROVED — Changes are test-only additions (162 insertions, 0 deletions across 2 new files), zero production code risk. `exp_log_contracts_test.go` pins exp/log/log2/log10 numeric contracts, inverse identities, homomorphism laws, and softmax shift-invariance. `parse_numeric_field_defaults_test.go` pins zero-value defaults for int/float/bool fields when omitted from JSON, with correct whitespace-normalized output matching. Both test files are well-structured, don't overlap with any of the 7 open automaintainer PRs, and the math/behavioral claims are hand-checkable. Regarding the off-objective flag from the intent check: the PR's Focus field is empty, but the changes constitute legitimate pinning tests that complement PR #450 (string-field parse contracts) — the numeric-field complement is a natural non-overlapping addition. CI is pending at review time; no blockers in the code itself.

**Comment by javimosch** [comment]
Approved by automaintainer review team — auto-merging
